components/script/dom/document.rs

```diff
@@ -376,8 +376,12 @@ impl Document {
     }
 
     // https://dom.spec.whatwg.org/#concept-document-url
-    pub fn url(&self) -> &Url {
-        &self.url
+    pub fn url(&self) -> Ref<Url> {
+        self.url.borrow()
```

Comment:
This can just use a `.clone()` for now, returning `Ref<Url>` isn't the scope of this PR.